lib/vsc/utils/missing.py

```diff
@@ -137,6 +137,23 @@ def star(self):
         """Return a new similar monoid."""
         return Monoid(self.null, self.mappend)
 
+class MonoidConcat(object):
```

Comment:
i don't understand the requirment/benefit for using monoids, but i think it is better that optimised `mappend` are simply supported in the `Monoid` class itself. eg. is you pass string `concat` as mappend, it uses this optimised way (by avodinh reduce)